### DISCUSSION TITLE
Is there any way to trigger a specific job (i.e. video converting) via CLI?

### DISCUSSION THREAD

**Discussion by rahilarious** [bpatrik/pigallery2#842] [Q&A] [answered]
I run pigallery2 without "Password protection" and as a LimitedGuest. Thanks for such a great app.

So everytime I upload some video, to transcode them, I need to enable Password protection and then run job from WEBUI's settings  and once that is done, turn off Password protection.

This work flow is cumbersome. If I could do run the job via CLI, it would be great.

**Comment by bpatrik** [comment]
There is a similar issue somewhere.

Ppl use curl or similar to do the http call that the app does from settings.

-- Sorry for being brief, sent from my phone.

**Reply by rahilarious** [reply]
I tried looking for that code which "app does from settings", but couldn't. May you please help me find that?

**Reply by bpatrik** [reply]
I would just open up the developer console then hit "transcode video" button and see what request does the app send and clone that. 

But If I recall correctly, if you disable password protection and the def user is not Admin you wont be able to trigger the job with curl either ( as that would be a security issue).

I would just set up a scheduled job whitin that app that runs every evening and transcodes what needs to be transcoded.

**Comment by rahilarious** [comment]
> I would just set up a scheduled job whitin that app that runs every evening and transcodes what needs to be transcoded.

That's exactly what I'm trying to do. Setup some kind of a cron job. But I'm not that familiar with JS. That's why I was asking for some CLI (bash) solution

**Reply by bpatrik** [reply]
Here is a related issue https://github.com/bpatrik/pigallery2/issues/520

**Reply by bpatrik** [reply, marked as answer]
The app has a built in scheuler I mean to use that: 
![kép](https://github.com/bpatrik/pigallery2/assets/7512182/fe8b962e-b3d6-49c3-b507-38bf397c4441)

using bash and cron will be very complex, you will need to emulate a browser and do a log in. If you do not have password protection enabled but the default user is not admin, its not possible to do.

**Reply by rahilarious** [reply]
This is amazing. I wish there was an option to run it periodically every 12 hours or shorter. But 24 hours still does the job. Thanks much. <3
